sources/api/schnauzer/src/helpers.rs

```diff
@@ -975,6 +975,86 @@ pub fn join_array(
     Ok(())
 }
 
+/// `join_nested_array` is used to join a nested array of scalar strings into a
+/// nested array of quoted, delimited strings. The delimiter must be specified.
+/// This handlebar template only supports array of array of strings.
+///
+/// # Example
+///
+/// Consider an array of values: `[[ "a", "b", "c" ], [ "d", "e" ]]` stored in a
+/// setting such as `settings.somewhere.foo-list`. In our template we can write:
+/// `{{ join_nested_array settings.somewhere.foo-list }}`
+///
+/// This will render `["a", "b", "c"], ["d", "e"]`.
+pub fn join_nested_array(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting join_nested_array helper");
+    let template_name = template_name(renderctx);
+    check_param_count(helper, template_name, 1)?;
+
+    let nested_array_param = get_param(helper, 0)?;
+    let nested_array =
+        nested_array_param
+            .as_array()
+            .with_context(|| error::JoinStringsWrongTypeSnafu {
+                expected_type: "array",
+                value: nested_array_param.to_owned(),
+                template: template_name,
+            })?;
+
+    let mut result = String::new();
+
+    for (i, array) in nested_array.iter().enumerate() {
+        if i > 0 {
```

Comment:
This is quite efficient, plus the tests give me confidence. As a future note, I think a quite "rust-y" way to accomplish this may have been to use iterators + [`Vec::join`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.join), e.g. something like:


```rust
let nested_array = nested_array_param.as_array().with_context(/* etc */)?;
let nested_array: Vec<String> = nested_array.into_iter()
    .map(|inner_array| {
        let inner_array = inner_array.as_array().with_context(/* etc */)?;
        let inner_array: Vec<String> = inner_array.into_iter()
            .map(|value| format!("\"{}\""), value.as_str().context(/* etc */))
            .collect::<Result<Vec<_>>>()?;
        Ok(inner_array.join(", "))
    })
    .collect::<Result<Vec<_>>>()?;
let result = nested_array.join(", ");

Reply:
Replaced join_nested_array with toml_encode as discussed.